stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -3318,16 +3318,23 @@ julia> blockdiag(sparse(2I, 3, 3), sparse(4I, 2, 2))
  ⋅  ⋅  ⋅  ⋅  4
 ```
 """
+function blockdiag(X::AbstractSparseMatrixCSC{Tv, Ti}...=spzeros(0,0)) where {Tv, Ti <: Integer}
```

Comment:
This default value makes the function change behavior for empty arguments. Currently, we have
```julia
julia> A = blockdiag()
0×0 SparseMatrixCSC{Union{},Int64} with 0 stored entries

julia> B = spzeros(0,0)
0×0 SparseMatrixCSC{Float64,Int64} with 0 stored entries
```
Changing that behavior may be for the better, but I'm not sure. This definitely requires a broader discussion, I'd say.

Reply:
I would rather not change the behavior for this pull request anyway; the goal here is to provide increased speed when dealing with CSC arrays with the same types for the index and values.

Created a new commit that fixes this (the output is now `SparseMatrixCSC{Union{},Int64}`) by explicitly spelling out the blockdiag() case, to make it clear what it does today.